### ISSUE TITLE
classify data error

### ISSUE THREAD

**Issue by ZeroAct** [ultralytics/ultralytics#1265]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

If  I don't specify model, it regards the data is not a classify dataset, even i queried `yolo classify train`.
I think it is confusing.

### Environment

- ultralytics==8.0.49

### Minimal Reproducible Example

does not work
```
yolo classify train data=imagenet100
```
does work
```
yolo train model=yolov8n-cls.pt data=imagenet100
yolo classify train model=yolov8n-cls.pt data=imagenet100
```

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
Thanks for the bug report! We're working on this type of task misalignment currently. 

This is handled in cfg/init file. I'll try to reproduce and assign model from task in case of missing model argument. 

Right now we have automations in place to handle the opposite, a cls model passed for training with no dataset will automatically be assigned imagenet100, but the reverse logic is not in place currently.

**Comment by ZeroAct** [comment]
I've made some change about this issue.
Can you check this? (Whenever you get some time!)

**Comment by glenn-jocher** [comment]
@ZeroAct this issue should now be resolved in #1263 commit https://github.com/ultralytics/ultralytics/pull/1263/commits/d9e7faca2307ee302cfe360da852ce36dc3011d3

Models will be now be assigned if missing from a task argument. Tested on:

```
yolo train task=classify data=imagenet100
```
